net/netutil/default_interface_portable.go

```diff
@@ -0,0 +1,61 @@
+package netutil
+
+import (
+	"errors"
+	"net"
+	"net/netip"
+)
+
+// DefaultInterfacePortable looks up the current default interface using a portable lookup method that
+// works on most systems with a BSD style socket interface.
+//
+// Returns the interface name and IP address of the default route interface.
+//
+// If the default cannot be determined, an error is returned.
+// Requires that there is a route on the system servicing UDP IPv4.
+func DefaultInterfacePortable() (string, netip.Addr, error) {
```

Comment:
Do we have something similar already in the tree? I have a vague memory of such a thing, but could be wrong.

Reply:
It used to be in net/interfaces, but it was merged into netmon which feels too dynamic & heavy for this purpose. The approach that they take is also quite complex, and I've been meaning to type this approach into some code somewhere for a while, as it's extremely useful and portable compared to parsing the routing tables as done elsewhere.